src/DataProtection/DataProtection/src/DataProtectionUtilityExtensions.cs

```diff
@@ -30,9 +30,9 @@ public static class DataProtectionUtilityExtensions
         /// </para>
         /// </remarks>
         [EditorBrowsable(EditorBrowsableState.Never)]
-        public static string GetApplicationUniqueIdentifier(this IServiceProvider services)
+        public static string? GetApplicationUniqueIdentifier(this IServiceProvider services)
         {
-            string discriminator = null;
+            string? discriminator = null;
```

Comment:
looks empty string was normalized to null before, if we want to make the discriminator non null, we could use "" instead of null?  But maybe not worth the hassle I guess.

Reply:
I don't want to change behavior in this PR unless it is a bug.